### PR TITLE
chore: add metrics for braze alias failures

### PR DESCRIPTION
**Pull request by utsabc**
## What are the changes introduced in this PR?

This PR adds metrics for braze identify flow where errors are observed due missing `userId`, as braze returns 200 with an error message in response body we want to capture that as metrics

## What is the related Linear task?

Resolves INT-2259

### CODE REVIEW

**Comment by devops-github-rudderstack** [comment]
Test report for this run is available at: https://test-integrations-dev.s3.amazonaws.com/integrations-test-reports/rudder-transformer/3452/test-report.html

File: src/v0/destinations/braze/util.js [hunk 1]
```diff
@@ -655,6 +655,46 @@ function getPurchaseObjs(message, config) {
...
+   *   "message": "success"
+   *   }
+   */
+
+  const { aliases_processed: aliasesProcessed, errors } = brazeResponse;
+  if (aliasesProcessed === 0) {
+    stats.increment('braze_alias_failure_count', { destination_id: destinationId });
+    const errorType = [];
+    if (Array.isArray(errors)) {
+      errors.forEach((error) => {
```
Comment (koladilip) [inline src/v0/destinations/braze/util.js:687]: map should be used for this

File: src/v0/destinations/braze/util.js [hunk 2]
```diff
@@ -655,6 +655,46 @@ function getPurchaseObjs(message, config) {
...
+        errorType.push(error.type);
+      });
+    }
+    stats.increment('braze_alias_failure_error_type', {
+      destination_id: destinationId,
+      alias_error_type: errorType.join(','),
```
Comment (koladilip) [inline src/v0/destinations/braze/util.js:693]: this might increase the cardinality so push individual error types as separate metrics

Comment (koladilip) [reply src/v0/destinations/braze/util.js:693]: avoid this metric and use logs instead

Comment (utsabc) [reply src/v0/destinations/braze/util.js:693]: addressed

Comment (utsabc) [reply src/v0/destinations/braze/util.js:693]: Screenshot 2024-06-10 at 2 02 39 PM

File: src/util/prometheus.js [hunk 3]
```diff
@@ -601,6 +601,18 @@ class Prometheus {
...
         labelNames: ['destination_id'],
       },
+      {
+        name: 'braze_alias_failure_error_count',
+        help: 'braze_alias_failure_error_count',
+        type: 'counter',
+        labelNames: ['destination_id'],
+      },
+      {
+        name: 'braze_alias_failure_error_type',
```
Comment (koladilip) [inline src/util/prometheus.js:611]: remove this

Comment (utsabc) [reply src/util/prometheus.js]: Addressed

**Review by ItsSudip — APPROVED** [review]

**Pull request merged**